components/eventing-controller/reconciler/subscription-nats/reconciler.go

```diff
@@ -125,7 +123,9 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	err = r.natsClient.DeleteSubscription(desiredSubscription)
 	if err != nil {
 		log.Error(err, "failed to delete subscriptions")
-		err = r.syncSubscriptionStatus(ctx, actualSubscription, false, err.Error())
+		if err := r.syncSubscriptionStatus(ctx, actualSubscription, false, err.Error()); err != nil {
+			return ctrl.Result{}, err
```

Comment:
the effect is correct, but imho the reason for the re-reconcile is off. here is seems to be reconciled as the subscriptionstatus could not be synced. But the original reason for reconciling again is that the deletion failed

Reply:
If the subscription status could not be saved, means that the status of the subscriptions previously saved is wrong. It makes the error analysis much difficult. Cause the reconciler should be idempotent, a new reconcile could perhaps solve the issue. This is let say a pattern... 
Of course, in the new call to reconcile(), the base action could succeed (in our case delete). 
From the perspective of reconciling calls, the error is important (!=nil) and not its type.